controllers/logicalvolume_controller.go

```diff
@@ -46,25 +48,26 @@ func (r *LogicalVolumeReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	log := crlog.FromContext(ctx)
 
 	lv := new(topolvmv1.LogicalVolume)
-	if err := r.Get(ctx, req.NamespacedName, lv); err != nil {
+	if err := logicalvolumeclient.Get(ctx, r, req.NamespacedName, lv); err != nil {
```

Comment:
I'm worried about the functions like `r.Get` being mistakenly used in the future after this PR is merged.
After this PR is merged, we must not use directly the `client`'s functions embedded in `LogicalVolumeReconciler`.
Instead, we need to use the wrapper functions like `logicalvolumeclient.Get`.
However, because the compiler does not complain about the direct use of client functions, preventing those mistakes seems difficult for me.

In my opinion, it is good if you could give the appropriate client object depending on whether the legacy mode is enabled or not at the following line. Then, we can use `client`'s functions directly.
https://github.com/topolvm/topolvm/blob/main/pkg/topolvm-node/cmd/run.go#L73

What do you think about this idea?

Reply:
I checked that you implemented the wrapped client. So I resolve this comment.